src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateCopy = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete stateCopy[keyToRemove];
+        }
+        break;
+      case 'clear':
+        stateCopy = {};
+        break;
+      default:
+        break;
```

Comment:
You can add "throw new Error" by default

Reply:
➕ , would be great to throw some error

Reply:
Done that! Thanks!!!